packages/instant/src/redux/analytics_middleware.ts

```diff
@@ -0,0 +1,45 @@
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import * as _ from 'lodash';
+import { Middleware } from 'redux';
+
+import { ETH_DECIMALS } from '../constants';
+import { AccountState } from '../types';
+import { analytics } from '../util/analytics';
+
+import { Action, ActionTypes } from './actions';
+
+import { State } from './reducer';
+
+export const analyticsMiddleware: Middleware = store => next => middlewareAction => {
+    const prevState = store.getState() as State;
```

Comment:
prevState and curState are the same?

Reply:
also what happens without the `as State` and `as Action`

Reply:
- it's a little funky, but calling `next(middlewareAction)` causes the getState() result to change.  I can add a comment for this
- without `as State` or `as Action` they are cast as `any`